auto-complete.js

```diff
@@ -117,50 +126,67 @@ var autoComplete = (function(){
             }
             that.parentElement.appendChild(that.sc);
 
-            live('autocomplete-suggestion', 'mouseleave', function(e){
+            live('autocomplete-suggestion', 'mouseleave', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
-                if (sel) setTimeout(function(){ sel.className = sel.className.replace('selected', ''); }, 20);
+                if (sel) setTimeout(function () { sel.className = sel.className.replace('selected', ''); }, 20);
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mouseover', function(e){
+            live('autocomplete-suggestion', 'mouseover', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
                 if (sel) sel.className = sel.className.replace('selected', '');
                 this.className += ' selected';
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mousedown', function(e){
+            live('autocomplete-suggestion', 'mousedown', function (e) {
                 if (hasClass(this, 'autocomplete-suggestion')) { // else outside click
                     var v = this.getAttribute('data-val');
-                    that.value = v;
-                    o.onSelect(e, v, this);
-                    that.sc.style.display = 'none';
+                    if (o.queryHistoryStorageName && e.target.parentElement.classList.contains('autocomplete-suggestion--local-remove-button')) {
```

Comment:
This check `e.target.parentElement.classList.contains` is fragile, you expect that it has an HTML element inside the button that is clicked, if this HTML structure is change this will break.

I suggest you to change the current HTML to be something like
```html
<div>
  <div class="autocomplete-suggestion autocomplete-suggestion--local">
    ...
  </div>
  <div class="autocomplete-suggestion--local-remove-button">
    ...
  </div>
</div>
```

and add a listener for the button click
```js
live('autocomplete-suggestion--local-remove-button', 'click', function (e) {
```

Reply:
I tried this solution, but doing this it would break the dropdown putting the _Remover_ button as a suggestion :confused: .
I will continue to implement some changes that to see if there is a way to implement it that way but I'm not sure it will work

Reply:
I tried to implement a suggestion but it did not find a way to generalize the case where we have local and non-local suggestions in the same dropdown. 
I looked further and other places in the code assume that the content of the autocomplete-suggestion class is an HTML. Like here:
```js
live('autocomplete-suggestion', 'mouseover', function (e) {
    var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
    if (sel) sel.className = sel.className.replace('selected', '');
    this.className += ' selected';
}, that.sc);
```
What we can do is place a check that the `e.target` is a document so that the code does not break. What do you think? @felipemsantana